model/histogram/float_histogram_test.go

```diff
@@ -1242,7 +1242,7 @@ func TestFloatHistogramAdd(t *testing.T) {
 				Sum:             1.234,
 				PositiveSpans:   []Span{{0, 2}, {3, 3}},
 				PositiveBuckets: []float64{5, 4, 2, 3, 6},
-				NegativeSpans:   []Span{{-9, 2}, {3, 2}},
+				NegativeSpans:   []Span{{-6, 2}, {1, 2}},
```

Comment:
Had to adjust this test a little - the input used to be somewhat invalid (some buckets were below `ZeroThreshold`), and this was failing the `b + a` case because `reconcileZeroBuckets` lacks symmetry (we cannot simply trim buckets of `other` histogram as it is not allowed to mutate):
https://github.com/prometheus/prometheus/blob/97f9ad9d317f1e227359d31de11f87fbc849e3f5/model/histogram/float_histogram.go#L733-L739

Not sure if this is something that needs to be solved though.

Reply:
My reasoning in the comment above is probably wrong: nothing in `reconcileZeroBuckets`/``trimBucketsInZeroBucket` applies in that test because both histograms have the same `ZeroThreshold` value (0.01). Might be some asymmetry in `addBuckets` then.

Reply:
What's valid is a bucket that is _partially_ covered by the zero threshold. So the absolute value of its lower limit is less than zeroThreshold, while the absolute value of its upper limit is greater than zeroThreshold. I agree that buckets that are completely within the zeroThreshold should be considered invalid.